array-manipulation-2/src/chunk.js

```diff
@@ -0,0 +1,33 @@
+/* exported chunk */
+
+/*
+- create storage for output
+- create storage for current array
+- push current array storage into the output
+- if the length in the last array in the output is less than size
+  - push current element to the last array in the output
+- if the length in the last array in the output is equal to size
+  - push new current array array into output
+  - push current element to the last array in the output
+- else,
+  - make output equal to array
+*/
+
+function chunk(array, size) {
+  var output = [];
+  var currentArray = [];
+  output.push(currentArray);
+  if (array.length === 0) {
+    return array;
+  }
+  for (var i = 0; i < array.length; i++) {
+    if (output[output.length - 1].length < size) {
+      output[output.length - 1].push(array[i]);
+    } else if (output[output.length - 1].length === size) {
+      currentArray = [];
+      output.push(currentArray);
+      output[output.length - 1].push(array[i]);
```

Comment:
This line is present in both blocks of your conditional. That means that it can simply be done below the conditional since it will happen in both cases. Because of this, one block would be empty, so you can make your `else if` here the primary `if` statement, and then do this `push` below that conditional.